addons/mail/static/src/messaging/entity/entity/entity.js

```diff
@@ -0,0 +1,800 @@
+            }
+            const relation = this.relations[relationName];
+            const storedRelationName = this.__getStoredRelationName(relationName);
+            const prevValue = entity[storedRelationName];
+            const value = relationValue === null
+                ? [...entity[storedRelationName]]
+                : relationValue instanceof Array
+                ? relationValue.map(e => e.isEntity ? e.localId: e)
+                : [relationValue.isEntity ? relationValue.localId : relationValue];
+            entity._write({
+                [storedRelationName]: entity[storedRelationName].filter(
+                    valueItem => !value.includes(valueItem)
+                ),
+            });
+            if (prevValue) {
+                const RelatedEntity = this.env.entities[relation.to];
+                for (const valueItem of value) {
+                    const related = RelatedEntity.get(valueItem);
+                    const storedRelatedRelationName =
+                        RelatedEntity.__getStoredRelationName(relation.inverse);
+                    related._write({
+                        [storedRelatedRelationName]: undefined,
+                    });
+                    if (relation.isCausal) {
+                        related.delete();
+                    }
+                }
+            }
+        }
+
+        /**
+         * @static
+         * @private
+         * @param {mail.messaging.entity.Entity} entity instance or class
+         * @param {Object} param1
+         * @param {string} param1.relationName
+         */
+        static __unlinkSingleOne2One(entity, { relationName }) {
+            if (!this.get(entity)) {
+                // Entity has already been deleted.
+                // (e.g. unlinking one of its reverse relation was causal)
+                return;
+            }
+            const relation = this.relations[relationName];
+            const storedRelationName = this.__getStoredRelationName(relationName);
+            const prevValue = entity[storedRelationName];
+            entity._write({ [storedRelationName]: undefined });
+            const RelatedEntity = this.env.entities[relation.to];
+            if (prevValue) {
+                const related = RelatedEntity.get(prevValue);
+                const storedRelatedRelationName =
+                    RelatedEntity.__getStoredRelationName(relation.inverse);
+                related._write({ [storedRelatedRelationName]: undefined });
+            }
+        }
+
+        /**
+         * @static
+         * @private
+         * @param {mail.messaging.entity.Entity} entity
+         * @param {Object} [data={}]
+         */
+        static __update(entity, data = {}) {
+            entity._update(data);
+        }
+
+    }
+
+    Object.assign(Entity, {
+        /**
+         * Registry containing data to make Entity classes. Entity classes should
+         * register themselves through static method `Entity.registerNewEntity()`.
```

Comment:
To check for later, but I don't like stuff registering themselves. There should be a "manager" registering only what it wants.

Reply:
This is the way if we want to manage creation of entities in different modules.